syscalls.txt

```diff
@@ -0,0 +1,83 @@
+# <stub name> <architecture> <syscall name> <parameters> <return type> <flags>
```

Comment:
Where does this file come from? Is it really easier to maintain this than the actual syscalls?

Reply:
> Where does this file come from? 

The idea is to write this once for each OS (`syscalls_linux.txt`, `syscalls_freebsd.txt`) and then have it auto-generate the stubs for every architecture.

> Is it really easier to maintain this than the actual syscalls?

Yes. The code generator understands some platform-specific quirks (eg. 32bit ARM/MIPS systems expect all the 64-bit parameters to be aligned on a even parameter number) or other crazyness (eg. on linux x86 systems all the socket syscalls go trough an extra indirection).

Reply:
IMO it's too "magic". For debugability and readability I'd prefer we just write it out.